lib/Sema/CSDiag.cpp

```diff
@@ -3443,15 +3443,37 @@ Expr *FailureDiagnosis::typeCheckChildIndependently(
       if (FT->isAutoClosure())
         convertType = FT->getResult();
 
-    // Replace archetypes and type parameters with UnresolvedType.
-    convertType = replaceTypeParametersWithUnresolved(convertType);
-    convertType = replaceTypeVariablesWithUnresolved(convertType);
-    
+    bool shouldNullify = false;
```

Comment:
Ok, maybe then just factoring this in its own method would be nice. Also please add a comment with the above explanation :-)